src/Tools/dotnet-monitor/Profiler/ProfilerEndpointInfoSourceCallbacks.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Profiler
+{
+    internal sealed class ProfilerEndpointInfoSourceCallbacks : IEndpointInfoSourceCallbacks
```

Comment:
After this PR, allow profiler to be loaded on demand or at startup as required by each feature that needs it.